crates/volta-core/src/error/mod.rs

```diff
@@ -45,6 +45,20 @@ impl VoltaError {
     pub fn kind(&self) -> &ErrorKind {
         &self.inner.kind
     }
+
+    /// Verifies if the inner source is `io::ErrorKind::NotFound`
+    pub fn is_not_found_error_kind(&self) -> bool {
```

Comment:
I'm a little wary of having a method on `VoltaError` that is only relevant for one specific kind of one specific source error. I like the idea in general of being able to possibly downcast the source, but I wonder if we can make it more generic and then provide a helper function (possibly in `volta_core::fs` for the specific "not found" case?

Something like:

```rust
pub fn caused_by<E, F>(&self, predicate: F) -> bool
where
    E: Error + 'static,
    F: FnOnce(&E) -> bool,
{
    match self.source().and_then(|e| e.downcast_ref::<E>()) {
        Some(err) => predicate(err),
        None => false,
    }
}
```
and
```rust
pub fn not_found(err: &io::Error) -> bool {
    err.kind() == io::ErrorKind::NotFound
}
```

At which point this calling this function could be turned from: `err.is_not_found_error_kind()` to `err.caused_by(not_found)`, which maintains the expressiveness while allowing us to re-use it in other spots where we might want to look up the sub-error.

Edit: After looking at the scope of other changes, I'm not sure if this will still be needed (though it may be inside the `from_file_if_exists` methods, I'm not sure on that). But if it is, keeping things more generic would be useful I think.

Reply:
Done